src/components/primitive/Icons/icon.component.tsx

```diff
@@ -0,0 +1,16 @@
+import { IconProps } from './icon.component.props';
+import { ReactSVG } from 'react-svg';
+import { ICONS } from './icons';
+import { SvgWrapper } from './icon.component.style';
+
+export const Icon = ({ icon, className, size = 14, color, ...rest }: IconProps) => (
```

Comment:
If there is a possibility that we might be using open source icon libraries where we don't have to deal with SVGs, we might need to name the component as Icon itself. In such a case, it's better to rename this component as SvgIcon.